core/src/main/java/cz/fi/muni/pv217/employee/management/EmployeeResource.java

```diff
@@ -0,0 +1,95 @@
+package cz.fi.muni.pv217.employee.management;
+
+import cz.fi.muni.pv217.employee.management.entity.Employee;
+import cz.fi.muni.pv217.employee.management.service.EmployeeService;
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.security.identity.SecurityIdentity;
+
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+
+@Path("/api/employee")
+@ApplicationScoped
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class EmployeeResource {
+    
+    @Inject
+    EmployeeService employeeService;
+
+    @Inject
+    SecurityIdentity identity;
+
+    @POST
+    @Path("/create")
+    @RolesAllowed("admin")
+    public Employee createEmployee(Employee employee) {
+        return employeeService.createEmployee(employee);
+    }
+
+    @PUT
+    @Path("/{id}/update")
+    @RolesAllowed("admin")
+    public Employee updateEmployee(@org.jboss.resteasy.annotations.jaxrs.PathParam long id, Employee employee) {
+        return employeeService.updateEmployee(id, employee);
+    }
+
+    @DELETE
+    @Path("/{id}/delete")
+    @RolesAllowed("admin")
+    public Response deleteEmployee(@org.jboss.resteasy.annotations.jaxrs.PathParam long id) {
+        Employee employee;
+
+        try {
+            employee = employeeService.deleteEmployee(id);
+
+            if (employee == null) {
+                return Response
+                        .status(Response.Status.INTERNAL_SERVER_ERROR)
+                        .entity("Cannot delete employee with id " + id)
+                        .build();
+            }
+        } catch (NotFoundException nfe) {
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(String.format("Employee with id %d not found.", id))
+                    .build();
+        }
+
+        return Response.ok(employee).build();
+    }
+
+    @GET
+    @RolesAllowed({"user","admin"})
```

Comment:
I dont understand role "user", isn't user anyone who is not admin? User should have access only to himself so you should get the current user from the authentication and limit your search only to him